packages/syft/src/syft/client/client.py

```diff
@@ -323,6 +329,213 @@ def get_client_type(self) -> type[SyftClient]:
+            "communication_protocol": communication_protocol,
+        }
+        content = self._make_get(self.routes.ROUTE_API.value, params=params)
+        obj = _deserialize(content, from_bytes=True)
+        obj.connection = self
+        obj.signing_key = credentials
+        obj.communication_protocol = communication_protocol
+        if self.proxy_target_uid:
+            obj.node_uid = self.proxy_target_uid
+        return cast(SyftAPI, obj)
+
+    def login(
+        self,
+        email: str,
+        password: str,
+    ) -> SyftSigningKey | None:
+        # TODO: Implement message proxy forwarding for gateway
+
+        credentials = {"email": email, "password": password}
+        response = self._make_post(self.routes.ROUTE_LOGIN.value, credentials)
+        obj = _deserialize(response, from_bytes=True)
+
+        return obj
+
+    def register(self, new_user: UserCreate) -> Any:
+        # TODO: Implement message proxy forwarding for gateway
+
+        data = _serialize(new_user, to_bytes=True)
+        response = self._make_post(self.routes.ROUTE_REGISTER.value, data=data)
+        response = _deserialize(response, from_bytes=True)
+        return response
+
+    def make_call(self, signed_call: SignedSyftAPICall) -> Any:
+        msg_bytes: bytes = _serialize(obj=signed_call, to_bytes=True)
+        # Since JSON expects strings, we need to encode the bytes to base64
+        # as some bytes may not be valid utf-8
+        # TODO: Can we optimize this?
+        msg_base64 = base64.b64encode(msg_bytes).decode()
+
+        rev_proxy_url = self.vld_reverse_proxy.with_path(
+            self.routes.ROUTE_API_CALL.value
+        )
+        forward_proxy_url = self.vld_forward_proxy.with_path(VEILID_PROXY_PATH)
+        json_data = {
+            "url": str(rev_proxy_url),
+            "method": "POST",
+            "dht_key": self.dht_key,
+            "data": msg_base64,
+        }
+        response = requests.post(  # nosec
+            url=str(forward_proxy_url),
+            json=json_data,
+        )
+
+        if response.status_code != 200:
+            raise requests.ConnectionError(
+                f"Failed to fetch metadata. Response returned with code {response.status_code}"
+            )
+
+        result = _deserialize(response.content, from_bytes=True)
+        return result
+
+    def __repr__(self) -> str:
+        return self.__str__()
+
+    def __str__(self) -> str:
+        res = f"{type(self).__name__}:"
+        res = res + f"\n DHT Key: {self.dht_key}"
+        res = res + f"\n Forward Proxy: {self.vld_forward_proxy}"
+        res = res + f"\n Reverse Proxy: {self.vld_reverse_proxy}"
+        return res
+
+    def __hash__(self) -> int:
+        return (
+            hash(self.proxy_target_uid)
+            + hash(self.dht_key)
+            + hash(self.vld_forward_proxy)
+            + hash(self.vld_reverse_proxy)
+        )
+
+    def get_client_type(self) -> type[SyftClient]:
```

Comment:
This seems to be a redundant method in all the different client files. Maybe we need to clean this up later.